kernel/src/vm/vmar/vm_mapping.rs

```diff
@@ -156,6 +158,74 @@ impl VmMapping {
     }
 }
 
+pub struct VmMappingRequest<'a> {
+    vm_space: &'a VmSpace,
+    preempt_guard: &'a DisabledPreemptGuard,
+    page_aligned_addr: usize,
+}
+
+#[orpc_trait]
+pub trait VmMappingPolicy {
+    fn get_cursor<'a>(
+        &self,
+        req: &VmMappingRequest<'a>,
+    ) -> core::result::Result<Result<(CursorMut<'a>, PagingLevel)>, RPCError>;
```

Comment:
You don't need this nesting. There asterinas Error type implements `From<RPCError>` so ORPC errors will by automatically converted.